src/ProcessManager.php

```diff
@@ -11,9 +11,10 @@ class ProcessManager
     /**
      * @param Process[] $processes
      * @param int $maxParallel
-     * @param int $poll
+     * @param int $poll microseconds
+     * @param Callable $callback takes 3 args: $type, $buffer, $process
      */
-    public function runParallel(array $processes, $maxParallel, $poll = 1000)
+    public function runParallel(array $processes, $maxParallel, $poll = 1000, $callback = null))
```

Comment:
Actually i have 4 phantomjs instances and n testcases. Each phantom has his own port and should only execute one test a time.
 So i must manipulate the command before when i know which phantom instance is free.

Actually i have three events before after and processing.